sdk/forms/src/protected-view/iframe/components/view.ts

```diff
@@ -44,49 +65,46 @@ const DisplayValue = component(
         span.innerText = String(value);
         container.appendChild(span);
         children.push(span);
-        span.addEventListener('mouseenter', (event) => {
-          sendToParent('mouseenter', {
-            path,
-            x: event.clientX,
-            y: event.clientY,
-          });
-        });
-        span.addEventListener('mouseleave', () => {
-          sendToParent('mouseleave', {
-            path,
-          });
-        });
-        span.addEventListener('click', () => {
-          if (clickToCopy) {
-            navigator.clipboard.writeText(String(value));
-          }
-          sendToParent('click', {
-            path,
-          });
-        });
+        container.addEventListener('mouseenter', mouseEnterHandler);
+        container.addEventListener('mouseleave', mouseLeaveHandler);
+        container.addEventListener('click', clickHandler);
```

Comment:
I think I prefer it being on the `span` and not the `container` because the container might also have a label and I want the event to trigger specifically for the value and not the label.

Reply:
The problem with a span is that it's declared inside switch case, hence it's not accessible outside of it. This is why I've replaced it with container from the outer scope. Though you're right and current behaviour is not correct, we want this behaviour only for values, not labels. I will move span outside of the switch then, can't see any other options

Reply:
I suggest just add `const cleanupFunctions: (() => void)[] = []` outside the switch.
The in the case do: cleanupFunctions.push(() => { ... })`
And in the `unmount` function do: `cleanupFunctions.forEach(f => f())`

Reply:
Makes sense, added cleanupFunctions and updated the logic